Classes/OldModel/Event.php

```diff
@@ -389,27 +389,45 @@ public function getPlacesWithCountry(): array
             return [];
         }
 
-        $countries = [];
-
-        $countryData = \Tx_Oelib_Db::selectMultiple(
-            'country',
-            'tx_seminars_sites LEFT JOIN ' .
-            'tx_seminars_seminars_place_mm ON tx_seminars_sites' .
-            '.uid = tx_seminars_seminars_place_mm.uid_foreign' .
-            ' LEFT JOIN tx_seminars_seminars ON ' .
-            'tx_seminars_seminars_place_mm.uid_local = ' .
-            'tx_seminars_seminars.uid',
-            'tx_seminars_seminars.uid = ' . $this->getUid() .
-            ' AND tx_seminars_sites.country <> ""' .
-            \Tx_Oelib_Db::enableFields('tx_seminars_sites'),
-            'country'
-        );
-
-        foreach ($countryData as $country) {
-            $countries[] = $country['country'];
-        }
+        $queryBuilder = self::getQueryBuilderForTable('tx_seminars_sites');
+        $countryData = $queryBuilder
+            ->select('country')
+            ->from('tx_seminars_sites')
+            ->from('tx_seminars_seminars_place_mm')
+            ->from('tx_seminars_seminars')
+            ->leftJoin(
+                'tx_seminars_sites',
+                'tx_seminars_seminars_place_mm',
+                'place_mm',
+                $queryBuilder->expr()->eq(
+                    'tx_seminars_sites.uid',
+                    $queryBuilder->quoteIdentifier('place_mm.uid_foreign')
+                )
+            )
+            ->leftJoin(
+                'tx_seminars_seminars_place_mm',
+                'tx_seminars_seminars',
+                'seminars',
+                $queryBuilder->expr()->eq(
+                    'tx_seminars_seminars_place_mm.uid_local',
+                    $queryBuilder->quoteIdentifier('seminars.uid')
+                )
+            )
+            ->where(
+                $queryBuilder->expr()->eq(
+                    'tx_seminars_seminars.uid',
+                    $queryBuilder->createNamedParameter($this->getUid(), \PDO::PARAM_INT)
+                ),
+                $queryBuilder->expr()->neq(
+                    'tx_seminars_sites.country',
+                    $queryBuilder->createNamedParameter('', \PDO::PARAM_STR)
+                )
+            )
+            ->groupBy('country')
+            ->execute()
+            ->fetchAll();
 
-        return $countries;
+        return array_column($countryData, 'country');
```

Comment:
Please use `getPlacesAsArray` to avoid code duplication. We then can use `array_colum` and `array_unique` on the result.

Reply:
The `getPlacesAsArray()` query does not exclude empty countries, so this method will return an empty string instead of an empty array. How do we want to handle this?

Reply:
We can use `array_filter` to remove the empty string (if there is any).